src/fireedge/src/modules/utils/string.js

```diff
@@ -0,0 +1,98 @@
+ *                                                                           *
+ * http://www.apache.org/licenses/LICENSE-2.0                                *
+ *                                                                           *
+ * Unless required by applicable law or agreed to in writing, software       *
+ * distributed under the License is distributed on an "AS IS" BASIS,         *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  *
+ * See the License for the specific language governing permissions and       *
+ * limitations under the License.                                            *
+ * ------------------------------------------------------------------------- */
+
+/**
+ * Upper case the first character of an input string.
+ *
+ * @param {string} input - Input string
+ * @returns {string} Input string modified
+ */
+export const upperCaseFirst = (input) =>
+  input?.charAt(0)?.toUpperCase() + input.substring(1)
+
+/**
+ * Transform into a lower case with spaces between words, then capitalize the string.
+ *
+ * @param {string} input - String to transform
+ * @returns {string} Sentence
+ * @example // "test-string" => "Test string"
+ * @example // "test_string" => "Test string"
+ * @example // "testString" => "Test string"
+ * @example // "TESTString" => "Test string"
+ */
+export const sentenceCase = (input) => {
+  const sentence = input
+    ?.replace(/[-_]([A-Za-z])/g, ' $1')
+    ?.replace(/([A-Z])([A-Z][a-z])/g, '$1 $2')
+    ?.replace(/([a-z])([A-Z])/g, '$1 $2')
+    ?.toLowerCase()
+
+  return upperCaseFirst(sentence)
+}
+
+/**
+ * Transform into a string with the separator denoted by the next word capitalized.
+ *
+ * @param {string} input - String to transform
+ * @returns {string} string
+ * @example // "test-string" => "testString"
+ * @example // "test_string" => "testString"
+ */
+export const camelCase = (input) =>
+  input
+    .toLowerCase()
+    .replace(/([-_\s][a-z])/gi, ($1) => $1.toUpperCase().replace(/[-_\s]/g, ''))
+
+/**
+ * Transform into a snake case string.
+ *
+ * @param {string} input - String to transform
+ * @returns {string} string
+ * @example // "test-string" => "test_string"
+ * @example // "testString" => "test_string"
+ * @example // "TESTString" => "test_string"
+ */
+export const toSnakeCase = (input) => sentenceCase(input).replace(/\s/g, '_')
+
+/**
+ * Converts the string value into a boolean.
+ *
+ * @param {string} str - String value.
+ * @returns {boolean} - If str is "yes" or 1 then returns true,
+ * in other cases, return false.
+ */
+export const stringToBoolean = (str) =>
+  ['yes', 'true'].includes(String(str).toLowerCase()) || +str === 1
+
+/**
+ * Replace < for &lt; and > for &gt;.
+ *
+ * @param {string} xmlString - The string with xml value
+ * @returns {string} - A string with the same value but with the replace characters
+ */
+export const transformXmlString = (xmlString = '') =>
+  xmlString.replace(/[<>&"']/g, function (c) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).